llvm/include/llvm/Telemetry/Telemetry.h

```diff
@@ -27,14 +27,14 @@ namespace telemetry {
 
 class Serializer {
 public:
-  virtual llvm::Error init() = 0;
+  virtual Error init() = 0;
   virtual void write(StringRef KeyName, bool Value) = 0;
   virtual void write(StringRef KeyName, int Value) = 0;
-  virtual void write(StringRef KeyName, size_t Value) = 0;
+  virtual void write(StringRef KeyName, unsigned long long Value) = 0;
```

Comment:
This is a slightly weird subset of types that you've provided overloads for. Could you explain why you've chosen these three specifically and no other fundamental types?

Reply:
These types are based on the set of data we need to collect in LLDB.
 - bool, int : basic types, nothing too strange
 -  size_t: this is needed to represent the size of a debug target and also for storing number of nanoseconds.
 - StringRef: binary paths, names, etc
 - map<string,string>: for storing misc data (eg., binary's ID fields, error messages, etc).

Reply:
Just to clarify, so you're suggesting adding additional e `write` overloads for `unsigned long`, `unsigned int`, `long` ?

Reply:
Yes, plus `long long`, i.e. you'd have 6 different integral overloads.

Reply:
Just repinging [my idea](https://godbolt.org/z/16cTva1fY) for redirecting everything to two overloads (for signed vs. unsigned). (Doing that because you seems to have converged on something similar for map types as well).

Reply:
> Just repinging [my idea](https://godbolt.org/z/16cTva1fY) for redirecting everything to two overloads (for signed vs. unsigned). (Doing that because you seems to have converged on something similar for map types as well).

Ah, that's a good idea. thanks!